FirebaseAdmin/FirebaseAdmin/Messaging/Util/AnalyticsLabelChecker.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Text.RegularExpressions;
+
+namespace FirebaseAdmin.Messaging.Util
+{
+    /// <summary>
+    /// Checker for analytics label.
+    /// </summary>
+    internal static class AnalyticsLabelChecker
+    {
+        private static string pattern = "^[a-zA-Z0-9-_.~%]{0,50}$";
+
+        /// <summary>
+        /// Checks anytytics labels an throw if not valid.
+        /// </summary>
+        /// <exception cref="ArgumentException">If analytics label does not match pattern.</exception>
+        /// <param name="analyticsLabel">Analytics label.</param>
+        public static void ValidateAnalyticsLabel(string analyticsLabel)
+        {
+            if (string.IsNullOrWhiteSpace(analyticsLabel))
```

Comment:
I think we ought to allow null values for AnalyticsLabel. In the future we will have other properties in `FcmOptions`, and it should be possible to specify them and leave AnalyticsLabel as null.